src/Lamar/IoC/Instances/InterceptingInstance.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Lamar.IoC.Frames;
+using LamarCodeGeneration.Model;
+
+namespace Lamar.IoC.Instances
+{
+    public class InterceptingInstance<T> : Instance
+    {
+        private readonly Instance _inner;
+        private readonly IActivationInterceptor<T> _interceptor;
+
+        public InterceptingInstance(Instance inner, IActivationInterceptor<T> interceptor)
+            : base(inner.ServiceType, inner.ImplementationType, inner.Lifetime)
+        {
+            _inner = inner;
+            _interceptor = interceptor;
+            Name = _inner.Name;
+        }
+
+        protected override IEnumerable<Instance> createPlan(ServiceGraph services)
+        {
+            _inner.CreatePlan(services);
+            foreach(var message in _inner.ErrorMessages)
+                ErrorMessages.Add(message);
+            return base.createPlan(services);
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            _inner.Reset();
+        }
+
+        public override Variable CreateVariable(BuildMode mode, ResolverVariables variables, bool isRoot) => new GetInstanceFrame(this).Variable;
+
+        public override Func<Scope, object> ToResolver(Scope topScope) => s => Resolve(topScope);
```

Comment:
`Resolve(topScope)` should be `Resolve(s)`.

Reply:
@jeremydmiller Just got the notification that you merged this, please see above. We had issues which was resolved by changing:
`public override Func<Scope, object> ToResolver(Scope topScope) => s => Resolve(topScope);`
to:
`public override Func<Scope, object> ToResolver(Scope topScope) => s => Resolve(s);`